### PR TITLE
qt5 5.5.0

### PR DESCRIPTION
**Pull request by beemaster**

### CODE REVIEW

**Comment by DomT4** [comment]
@BrewTestBot test this please

**Comment by tdsmith** [comment]
@BrewTestBot test this please

**Comment by tdsmith** [comment]
This may need to be more involved; tests are failing.

Interesting note in [new features](https://wiki.qt.io/New_Features_in_Qt_5.5): "New SSL back-end for iOS and OS X based on Secure Transport. Note that in Qt 5.6 this will become the default SSL: back-end on these platforms." It would be nice to turn that on.

**Comment by DomT4** [comment]
>  It would be nice to turn that on.

:+1: :+1: :+1:.

**Comment by beemaster** [comment]
Looks like the reason for failing build is [QTBUG-46833](https://bugreports.qt.io/browse/QTBUG-46833). Probably worth waiting for 5.5.1?

**Comment by DomT4** [comment]
IMO, It's a one line patch, I'd be happy to use it until the next release to get this in.

**Comment by beemaster** [comment]
I added the patch to formula.

**Comment by luzpaz** [comment]
Brewbot build fails: 
Yosemite:
```
==> brew test --verbose qt5						 FAILED
Testing qt5
==> /usr/local/Cellar/qt5/5.5.0/bin/qmake -project
Error: qt5: failed
execution expired
/usr/local/Library/Homebrew/formula.rb:868:in `gets'
/usr/local/Library/Homebrew/formula.rb:868:in `block in system'
/usr/local/Library/Homebrew/formula.rb:854:in `open'
/usr/local/Library/Homebrew/formula.rb:854:in `system'
/usr/local/Library/Formula/qt5.rb:127:in `block in <class:Qt5>'
/usr/local/Library/Homebrew/formula.rb:797:in `block in run_test'
/usr/local/Library/Homebrew/extend/fileutils.rb:17:in `mktemp'
/usr/local/Library/Homebrew/formula.rb:793:in `run_test'
/usr/local/Library/Homebrew/test.rb:28:in `block in <main>'
/usr/local/Library/Homebrew/test.rb:27:in `<main>'
==> brew fetch --retry pyqt5						 PASSED
==> brew install pyqt5							 PASSED
==> brew test --verbose pyqt5						 FAILED
Testing pyqt5
==> pyuic5 --version
Python User Interface Compiler 5.4.2 for Qt version 5.4.2
==> pylupdate5 -version
dyld: Library not loaded: @rpath/QtCore.framework/Versions/5/QtCore
  Referenced from: /usr/local/opt/qt5/lib/QtXml.framework/Versions/5/QtXml
  Reason: image not found
Error: pyqt5: failed
Failed executing: pylupdate5 -version
/usr/local/Library/Homebrew/formula.rb:897:in `block in system'
/usr/local/Library/Homebrew/formula.rb:854:in `open'
/usr/local/Library/Homebrew/formula.rb:854:in `system'
/usr/local/Library/Formula/pyqt5.rb:60:in `block in <class:Pyqt5>'
/usr/local/Library/Homebrew/formula.rb:797:in `block in run_test'
/usr/local/Library/Homebrew/extend/fileutils.rb:17:in `mktemp'
/usr/local/Library/Homebrew/formula.rb:793:in `run_test'
/usr/local/Library/Homebrew/test.rb:28:in `block in <main>'
/System/Library/Frameworks/Ruby.framework/Versions/2.0/usr/lib/ruby/2.0.0/timeout.rb:66:in `timeout'
/usr/local/Library/Homebrew/test.rb:27:in `<main>'
==> brew fetch --retry qbs						 PASSED
==> brew install qbs							 PASSED
==> brew test --verbose qbs						 FAILED
Testing qbs
==> /usr/local/Cellar/qbs/1.4.1/bin/qbs setup-toolchains --detect --settings-dir /private/tmp/qbs20150704-50087-cn6qnn
dyld: Library not loaded: @rpath/QtCore.framework/Versions/5/QtCore
  Referenced from: /usr/local/opt/qt5/lib/QtScript.framework/Versions/5/QtScript
  Reason: image not found
Error: qbs: failed
Failed executing: /usr/local/Cellar/qbs/1.4.1/bin/qbs setup-toolchains --detect --settings-dir /private/tmp/qbs20150704-50087-cn6qnn
/usr/local/Library/Homebrew/formula.rb:897:in `block in system'
/usr/local/Library/Homebrew/formula.rb:854:in `open'
/usr/local/Library/Homebrew/formula.rb:854:in `system'
/usr/local/Library/Formula/qbs.rb:42:in `block in <class:Qbs>'
/usr/local/Library/Homebrew/formula.rb:797:in `block in run_test'
/usr/local/Library/Homebrew/extend/fileutils.rb:17:in `mktemp'
/usr/local/Library/Homebrew/formula.rb:793:in `run_test'
/usr/local/Library/Homebrew/test.rb:28:in `block in <main>'
/System/Library/Frameworks/Ruby.framework/Versions/2.0/usr/lib/ruby/2.0.0/timeout.rb:66:in `timeout'
/usr/local/Library/Homebrew/test.rb:27:in `<main>'
```

**Comment by DomT4** [comment]
We finally have Yosemite build success, although the build on all three machines was a good ~25 minutes quicker than usual, which is a little odd.

Getting a lot of test failure we need to look into though:

`qt5` -
```
==> /usr/local/Cellar/qt5/5.5.0/bin/qmake -project
Error: qt5: failed
execution expired
```

`pyqt5` -

```
Testing pyqt5
==> pyuic5 --version
Python User Interface Compiler 5.4.2 for Qt version 5.4.2
==> pylupdate5 -version
dyld: Library not loaded: @rpath/QtCore.framework/Versions/5/QtCore
  Referenced from: /usr/local/opt/qt5/lib/QtXml.framework/Versions/5/QtXml
  Reason: image not found
Error: pyqt5: failed
Failed executing: pylupdate5 -version
```

`qbs` - 

```
Testing qbs
==> /usr/local/Cellar/qbs/1.4.1/bin/qbs setup-toolchains --detect --settings-dir /private/tmp/qbs20150704-50087-cn6qnn
dyld: Library not loaded: @rpath/QtCore.framework/Versions/5/QtCore
  Referenced from: /usr/local/opt/qt5/lib/QtScript.framework/Versions/5/QtScript
  Reason: image not found
Error: qbs: failed
```

**Comment by DomT4** [comment]
Heh. You can't untar the bottle using `bsdtar`. You can with `gnutar` but only to this point:

```
qt5/5.5.0/lib/QtWebEngineCore.framework/Versions/5/Libraries/
qt5/5.5.0/lib/QtWebEngineCore.framework/Versions/5/QtWebEngineCore
gzip: (stdin): unexpected end of file
gtar: Unexpected EOF in archive
gtar: Unexpected EOF in archive
gtar: Error is not recoverable: exiting now
```

[This is everything in the tarball up to that error](https://gist.github.com/DomT4/4d45ef0efcac8a714151).

**Comment by tdsmith** [comment]
The patch should only affect --with-developer; there's other weirdness.

**Comment by jakepetroules** [comment]
@DomT4 As of Qt 5.5, rpath support is turned on by default for OS X. The frameworks have rpath-based install names, and no LC_RPATHs.

Not sure what Homebrew's policy is on rpath but if you need it off there's a configure option for it (-no-rpath) and at this point it might be the simplest thing to do.

**Comment by DomT4** [comment]
Apologies, I went to bed and left `qt5` to build locally to see whether the failure was reproducible here. It isn't, I can pull this PR, build it, bottle it, untar the bottle as desired, and install and use the bottle.

```
==> Summary
🍺  /usr/local/Cellar/qt5/5.5.0: 7177 files, 426M, built in 219.6 minutes
```

So I'm not quite sure why the test-bot would fail on all three platforms, especially since I'm using the same command essentially as the test-bot does to produce bottles. [Full build logs here](https://gist.github.com/anonymous/d2047f4c8db4e58adc15).

@mikemcquaid Does the bot do anything that could break rpath usage on the bot exclusively?

**Comment by DomT4** [comment]
Since the bot is nice and quiet at the moment I've fired off #41367 to see whether the `rpath` changes are the failure.

**Comment by mikemcquaid** [comment]
Sorry, killed this as we need to build a bunch of other PRs right now.

File: Library/Formula/qt5.rb [hunk 1]
```diff
@@ -12,9 +12,15 @@ def message; <<-EOS.undent
...
-  sha256 "8c6d070613b721452f8cffdea6bddc82ce4f32f96703e3af02abb91a59f1ea25"
+  url "https://download.qt.io/official_releases/qt/5.5/5.5.0/single/qt-everywhere-opensource-src-5.5.0.tar.xz"
+  mirror "https://www.mirrorservice.org/sites/download.qt-project.org/official_releases/qt/5.5/5.5.0/single/qt-everywhere-opensource-src-5.5.0.tar.xz"
+  sha256 "7ea2a16ecb8088e67db86b0835b887d5316121aeef9565d5d19be3d539a2c2af"
+
+  # Apple's 3.6.0svn based clang doesn't support -Winconsistent-missing-override
+  # https://bugreports.qt.io/browse/QTBUG-46833
+  # This is fixed in 5.5 branch and below patch should be removed
+  # when this formula is updated to 5.5.1
+  patch :DATA
```
Comment (mikemcquaid) [inline Library/Formula/qt5.rb]: Please submit this to upstream's Gerrit/Code Review tool.

**Comment by luzpaz** [comment]
@mikemcquaid  The patch was missing from that annotation. I'll see what I can do to submit.

**Comment by luzpaz** [comment]
```diff
+diff --git a/qtbase/src/corelib/global/qcompilerdetection.h b/qtbase/src/corelib/global/qcompilerdetection.h
+index 7ff1b67..060af29 100644
+--- a/qtbase/src/corelib/global/qcompilerdetection.h
++++ b/qtbase/src/corelib/global/qcompilerdetection.h
+@@ -155,7 +155,7 @@
+ /* Clang also masquerades as GCC */
+ #    if defined(__apple_build_version__)
+ #      /* http://en.wikipedia.org/wiki/Xcode#Toolchain_Versions */
+-#      if __apple_build_version__ >= 6020049
++#      if __apple_build_version__ >= 7000053
+ #        define Q_CC_CLANG 306
+ #      elif __apple_build_version__ >= 6000051
+ #        define Q_CC_CLANG 305
```

**Comment by mikemcquaid** [comment]
@luzpaz Right, someone needs to submit that upstream.

**Comment by luzpaz** [comment]
Submitted upstream: https://bugreports.qt.io/browse/QTBUG-47130

**Comment by mikemcquaid** [comment]
@luzpaz can you submit to https://codereview.qt-project.org/?

**Comment by luzpaz** [comment]
@mikemcquaid re: codereview 
sorry, the instructions are a little too involved for me to do that right now. 

Did get a response back from a Qt dev:
https://github.com/Homebrew/homebrew/pull/41252#discussion-diff-34255151
He wants to know in more depth what the code does so he can ascertain a commit message

**Comment by luzpaz** [comment]
> You have to tell me why.

>In other words, I need a commit message that explains what this is fixing and why this is the correct solution.

[Link](https://bugreports.qt.io/browse/QTBUG-47130?focusedCommentId=285980&page=com.atlassian.jira.plugin.system.issuetabpanels:comment-tabpanel#comment-285980)

**Comment by mikemcquaid** [comment]
@beemaster Can you submit this patch upstream instead? If it's already been committed can you use the commit URL as the `patch` url

**Comment by beemaster** [comment]
@mikemcquaid I am a bit confused, I don't think this should be upstreamed. This is a known bug and is already fixed in qt 5.5 branch. It just did not make it's way to qt 5.5.0. But it will be included in qt 5.5.1.

**Comment by mikemcquaid** [comment]
> @mikemcquaid I am a bit confused, I don't think this should be upstreamed. This is a known bug and is already fixed in qt 5.5 branch. It just did not make it's way to qt 5.5.0. But it will be included in qt 5.5.1

if it's already been committed can you use the commit URL as the patch url, thanks.

**Comment by luzpaz** [comment]
Should I close [QTBUG-47130](https://bugreports.qt.io/browse/QTBUG-47130] then?

**Comment by mikemcquaid** [comment]
@luzpaz Looks like it's probably a duplicate of https://bugreports.qt.io/browse/QTBUG-46833

**Pull request closed without merging**